kolibri/auth/test/test_api.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+import factory
+
+from django.core.urlresolvers import reverse
+
+from rest_framework.test import APITestCase
+
+from .. import models
+
+
+class FacilityFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = models.Facility
+
+    name = factory.Sequence(lambda n: "Rock N' Roll High School #%d" % n)
+
+
+class FacilityUserFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = models.FacilityUser
+
+    facility = factory.SubFactory(FacilityFactory)
+    username = factory.Sequence(lambda n: 'user%d' % n)
+    password = factory.PostGenerationMethodCall('set_password', 'password')
+
+
+class FacilityApiTestCase(APITestCase):
+
+    def setUp(self):
+        self.password = 'abc123'
+        self.facility1 = FacilityFactory.create()
+        self.facility2 = FacilityFactory.create()
+        self.user1 = FacilityUserFactory.create(facility=self.facility1)
+        self.user2 = FacilityUserFactory.create(facility=self.facility2)
+
+    def test_sanity(self):
+        self.assertTrue(self.client.login(username=self.user1.username, password="password", facility=self.facility1))
+
+    def test_facility_user_can_get_detail(self):
+        self.client.login(username=self.user1.username, password="password", facility=self.facility1)
+        response = self.client.get(reverse('facility-detail', kwargs={'pk': self.facility1.pk}),
+                                   format='json')
+        # .assertDictContainsSubset checks that the first argument is a subset of the second argument
+        self.assertDictContainsSubset({
+            'name': self.facility1.name,
+        }, dict(response.data))
+
+    def test_anonymous_user_gets_empty_list(self):
+        response = self.client.get(reverse('facility-list'), format='json')
+        self.assertEqual(response.data, [])
```

Comment:
I could see the value in that, especially in the case of anon users (in case a logout happened). But I don't know that there's a systematic way (with the current system) on a list view to know the difference between there just being no existing records readable by a user (empty list), and there being no *way* for there to be records of this kind readable by the current user (403). Would love your help in thinking this through -- going to start by laying out a few examples.

- If a DeviceOwner does `GET /api/facility/`, but no facilities exist on the device, we probably don't want to return a 403, just an empty list. It's not that they're not allowed to ask, it's just that there aren't any.
- Similarly, if there are facility users for Facility A but not for Facility B, then when a coach for Facility B does `GET /api/facilityuser/`, it would seem to make sense to return an empty list, not a 403. They have read permission for a subset of facility users, it's just that their subset happens to be empty at the moment.
- The example shown in the test here is where it becomes more of a gray area for me. Similar to the last examples, we are filtering down the queryset to only those objects readable by the current (in this case, anonymous) user. While in this case (and probably the majority of cases) this list would be empty for an anon user, it might not always be -- there could be some models we want to allow an anonymous user to view.

So, I guess we have three main options:
- Always throw a 403 for anonymous requests
- Leave it like it is (on list views, just filter down to a readable subset, which for some users might never be anything but empty)
- Adapt the permissions system to specify table-level (non-object-level) permissions. If we see a strong use case, we can look at what that would take.

Let me know what you think! Thanks!

(Note that requests for a specific object/detail view already throw a 403 if unauthorized; this only applies to list views)